source/common/ssl/context_impl.cc

```diff
@@ -44,11 +46,13 @@ ContextImpl::ContextImpl(ContextManagerImpl& parent, Stats::Scope& scope,
   RELEASE_ASSERT(rc == 1);
 
   if (!SSL_CTX_set_strict_cipher_list(ctx_.get(), config.cipherSuites().c_str())) {
+    ERR_print_errors_fp(stdout);
```

Comment:
Instead of printing to stdout, please grab the error string and then use it when throwing the exception, e.g.:

```
throw EnvoyException(
          fmt::format("Failed to initialize cipher suites {}: {}", config.cipherSuites(), error));
```

Same elsewhere.

Reply:
Why can't we put the error trace in the exception? That way we could eventually report it all the way back to management server in a ACK/NACK `DiscoveryRequest`?

Reply:
Oh, forget what I just said, I didn't read @mattklein123's comment before hitting keyboard buttons :P